src/dev/i18n/utils.js

```diff
@@ -126,6 +128,32 @@ export function createParserErrorMessage(content, error) {
   return `${error.message}:\n${context}`;
 }
 
+/**
+ * Recursively extracts all references from ICU message ast.
+ *
+ * Example: `'Removed tag {tag} from {assignmentsLength, plural, one {beat {beatName}} other {# beats}}.'`
+ *
+ * @param {any} node
+ * @param {Set<string>} keys
+ */
+function extractValueReferencesFromIcuAst(node, keys = new Set()) {
+  if (Array.isArray(node.elements)) {
+    for (const element of node.elements.filter(element => element.type === ARGUMENT_ELEMENT_TYPE)) {
+      keys.add(element.id);
+
+      if (element.format && Array.isArray(element.format.options)) {
```

Comment:
__nit:__ please add a comment explaining what is `format` and what is `format` with `options` since based on this `if` you can have `format` with and without `options`.

Reply:
@azasypkin `intl-messageformat-parser` has no documentation or typings. I check `options` just for safety, because I'm not sure that there is no case when `format` has no options.